### ISSUE TITLE
Deploying in Bluemix with HSBN plan - need any changes?

### ISSUE THREAD

**Issue by likepunk** [IBM-Blockchain/car-lease-demo#70]
Hello!

I tried to switch Bluemix's blockchain services from ibm-blockchain-5-prod to HSBN(ibm-blockchain-plan-z2-prod), but it seems it doesn't work with HSBN. Do I have to change hardcoded label in the code to 'ibm-blockchain-plan-z2-prod'?  

Is there anything needs to be modified to work with HSBN?

Thanks in advance,

Alex

**Comment by likepunk** [comment]
I also tried to deploy with local version of node app - and got following errors. I guess there are additional credentials needs to be added to the hyperledger fabric server - how do I do  it with HSBN or regular Bluemix blockchain service with local application?

### 

[!] Running locally with bluemix fabric
peer0: grpcs://a78024a4-b0b7-4b90-829f-a218300e6909_vp0-discovery.zone.blockchain.ibm.com:30303
peer1: grpcs://a78024a4-b0b7-4b90-829f-a218300e6909_vp1-discovery.zone.blockchain.ibm.com:30303
peer2: grpcs://a78024a4-b0b7-4b90-829f-a218300e6909_vp2-discovery.zone.blockchain.ibm.com:30303
peer3: grpcs://a78024a4-b0b7-4b90-829f-a218300e6909_vp3-discovery.zone.blockchain.ibm.com:30303
membersrvc: grpcs://a78024a4-b0b7-4b90-829f-a218300e6909_ca-discovery.zone.blockchain.ibm.com:30303
eventhub: grpcs://undefined:undefined
Server Up
INFO    Startup complete on port        8080

INFO    Startup Enrolled registrar

INFO    Startup Set registrar

INFO    Startup Failed to enroll DVLA using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

{ Error: Signature verification failed.
    at /root/ibm/nodev6/carlease/car-lease-demo/node_modules/grpc/src/node/src/client.js:417:17 code: 2, metadata: Metadata { _internal_repr: {} } }
ERROR   Startup OUTPUT: {"code":2,"metadata":{"_internal_repr":{}}}
INFO    Startup Failed to enroll ScrapIt_UK using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Aston_Scrap_Centre using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Cray_Bros_London_Ltd using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Regionwide_Vehicle_Contracts using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Every_Car_Leasing using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll LeaseCan using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Milescape using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Beechvale_Group using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Anthony_O_Dowd using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Jaguar_Land_Rover using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Andrew_Hurt using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Alfa_Romeo using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Joe_Payne using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Toyota using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

INFO    Startup Failed to enroll Viewers_Alfa_Romeo using HFC. Error: {"code":2,"metadata":{"_internal_repr":{}}}

**Comment by 14gracel** [comment]
The error you are seeing when trying to deploy locally looks like a problem with the certificate you are using. You should save the HSBN certificate to `Chaincode/src/vehicle_code` and replace the current `certificate.pem`. 

And regarding the hard coded service name, for now you will have to change the hard coded name to ibm-blockchain-plan-z2-prod

**Comment by likepunk** [comment]
Okay @14gracel  thanks.. I tried to put cert.pem(what I got from registrar/ecert for WebAppAdmin) in that directory...didn't work. I  also put it under /certs/peer/cert.pem as app.js pointed for the path in below. Should I change also certificate var?


// Connecting to TLS enabled peers requires a certificate
var certificate = fs.readFileSync('us.blockchain.ibm.com.cert'); // TODO should download using service credentials

// Deploying chaincode requires us to know a path to a certificate on the peers :(
var certificate_path = '/certs/peer/cert.pem'; // TODO this should be available in the service credentials

**Comment by 14gracel** [comment]
The cert that you should save to the .pem file is the one that you get directly from the HSBN bluemix service. in the VCAP_SERVICES object (which is part of the bluemix service), you will see a key called `cert_path` which contains a url to a certificate. Download that certificate, and replace the current certificate.pem with the downloaded certificate, and rename it. I hope that helps

**Comment by likepunk** [comment]
Thanks @14gracel for another feedback!

I no longer have HSBN access so I changed to local docker-compose fabric(v0.6). What certification should I use if I want to use local fabric? I generated ecert for WebAppAdmin, but it seems it is not the cert it was expecting.

    peerClient: Client { '$channel': Channel {} } } ]
chain_setup.js Getting registrar: WebAppAdmin
app.js: Server Up - 10.100.0.148:3002
chain_setup.js successfully got registrar. Enrolling with secret: DJY27pEnl16d
chain_setup.js Successfully enrolled registrar
chain_setup.js using registrar to deploy chaincode
chain_setup.js Deploying commercial paper chaincode as: WebAppAdmin
chain_setup.js Deployment request: {"fcn":"init","args":["a","100"],"chaincodePath":"chaincode/","certificatePath":"/certs/peer/cert_test_WebAppAdmin.pem"}
E1216 12:59:52.873854186   16884 ssl_transport_security.c:945] Handshake failed with fatal error SSL_ERROR_SSL: error:1408F10B:SSL routines:SSL3_GET_RECORD:wrong version number.
E1216 12:59:52.873966961   16884 handshake.c:128]            Security handshake failed: {"created":"@1481911192.873946486","description":"Handshake failed","file":"../src/core/lib/security/transport/handshake.c","file_line":264,"tsi_code":10,"tsi_error":"TSI_PROTOCOL_FAILURE"}
E1216 12:59:52.877280423   16884 ssl_transport_security.c:945] Handshake failed with fatal error SSL_ERROR_SSL: error:1408F10B:SSL routines:SSL3_GET_RECORD:wrong version number.
E1216 12:59:52.877306016   16884 handshake.c:128]            Security handshake failed: {"created":"@1481911192.877294861","description":"Handshake failed","file":"../src/core/lib/security/transport/handshake.c","file_line":264,"tsi_code":10,"tsi_error":"TSI_PROTOCOL_FAILURE"}
chain_setup.js Failed to deploy chaincode: EventTransactionError {
  error:
   { Error
       at /root/ibm/node/cpdemo_local/cp-web/node_modules/grpc/src/node/src/client.js:417:17 code: 14, metadata: Metadata { _internal_repr: {} } },
  msg: 'Error' }
chain_setup.js chaincode deployment failed: undefined
app.js: Chain setup failed: EventTransactionError {
  error:
   { Error
       at /root/ibm/node/cpdemo_local/cp-web/node_modules/grpc/src/node/src/client.js:417:17 code: 14, metadata: Metadata { _internal_repr: {} } },
  msg: 'Error' }

/root/ibm/node/cpdemo_local/cp-web/app.js:226
            throw error;



cat cert_test_WebAppAdmin.pem
-----BEGIN+CERTIFICATE-----
MIIBtjCCAV2gAwIBAgIBATAKBggqhkjOPQQDAzAxMQswCQYDVQQGEwJVUzEUMBIG
A1UEChMLSHlwZXJsZWRnZXIxDDAKBgNVBAMTA2VjYTAeFw0xNjEyMTYxNzUxMjla
Fw0xNzAzMTYxNzUxMjlaMEcxCzAJBgNVBAYTAlVTMRQwEgYDVQQKEwtIeXBlcmxl
ZGdlcjEiMCAGA1UEAwwZV2ViQXBwQWRtaW5caW5zdGl0dXRpb25fYTBZMBMGByqG
SM49AgEGCCqGSM49AwEHA0IABH0acVeEen1DFp08MyPF%2BCSSCXLdL0QCk5tm6U0C
DwjrLk5x8ZCfD5aLMeC6Ws0Gk0A05ZmBUhly6hQcnB%2BM85%2BjUDBOMA4GA1UdDwEB
%2FwQEAwIHgDAMBgNVHRMBAf8EAjAAMA0GA1UdDgQGBAQBAgMEMA8GA1UdIwQIMAaA
BAECAwQwDgYGUQMEBQYHAQH%2FBAExMAoGCCqGSM49BAMDA0cAMEQCIFvYdbzat%2BBp
JjQaJEzvQnkfgpcVQB7CYqGEKIeMN1lVAiAfwxf5LKJPSSGUHT8sGrz1GFRI6lAv%0AjibpACBcdvBeZg%3D%3D
-----END+CERTIFICATE-----
